app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -52,15 +53,24 @@ class PatientSearchResultItemView(
     renderPatientNameAgeAndGender(name)
     renderPatientAddress(model.address)
     renderPatientDateOfBirth(model.dateOfBirth)
-    val number = highlightPatientPhoneNumber(searchQuery, model)
-    renderPatientPhoneNumber(number)
+    highlightPatientPhoneNumber(searchQuery, model)
+
     renderVisited(model.lastSeen)
     renderLastSeen(model.lastSeen, currentFacilityId)
   }
 
-  private fun highlightPatientPhoneNumber(searchQuery: String, patientSearchResult: PatientSearchResultViewModel): PhoneNumber {
-    val canHighlightNumber = searchQuery.isNotBlank()
-        && patientSearchResult.phoneNumber.toString().contains(searchQuery)
+  private fun highlightPatientPhoneNumber(searchQuery: String, patientSearchResult: PatientSearchResultViewModel) {
+    if (patientSearchResult.phoneNumber != null) {
+      val highlightNumber = searchQuery.isNotBlank()
+          && patientSearchResult.phoneNumber.toString().contains(searchQuery)
+      val number = canHighlightNumber(highlightNumber, patientSearchResult, searchQuery)
+      renderPatientPhoneNumber(number)
+    } else {
+      phoneNumberContainer.visibility = View.GONE
```

Comment:
We are already setting the phone number container visibility in the renderPhoneNumber method. Why are we doing this again here? 🤔 

Also, why did we remove the `renderPatientPhoneNumber` method?

Reply:
We are checking the visibility before we highlight or show the container. We did not remove `renderPatientPhoneNumber` method. We're just called it when the phoneNumber is not null.